@PaarthShah The tests are failing with `0.22.0`. Would you mind taking a look?
```py
(venv-311) $ pytest tests/components/matrix/
ImportError while loading conftest '/.../tests/components/matrix/conftest.py'.
tests/components/matrix/conftest.py:8: in <module>
    from nio import (
../venv-311/lib/python3.11/site-packages/nio/__init__.py:9: in <module>
    from .client import *
../venv-311/lib/python3.11/site-packages/nio/client/__init__.py:3: in <module>
    from .base_client import Client, ClientConfig
../venv-311/lib/python3.11/site-packages/nio/client/base_client.py:35: in <module>
    from ..crypto import ENCRYPTION_ENABLED, DeviceStore, OlmDevice, OutgoingKeyRequest, Sas
E   ImportError: cannot import name 'Sas' from 'nio.crypto' (/.../venv-311/lib/python3.11/site-packages/nio/crypto/__init__.py)
```

I believe it's because `Sas` is imported unconditionally here even though it's only available with the extra dependency which Home Assistant doesn't use. So I'm seeing an import error.
https://github.com/poljar/matrix-nio/blob/0.22.0/nio/client/base_client.py#L35

AFAICT it's only used in type annotations and `from __future__ import annotations` is already set. It should be enough to move the `Sas` import into an `if TYPE_CHECKING` block.

--- a/homeassistant/components/matrix/manifest.json
+++ b/homeassistant/components/matrix/manifest.json
@@ -5,5 +5,5 @@
   "documentation": "https://www.home-assistant.io/integrations/matrix",
   "iot_class": "cloud_push",
   "loggers": ["matrix_client"],
-  "requirements": ["matrix-nio==0.22.0", "Pillow==10.0.1"]
+  "requirements": ["matrix-nio==0.22.1", "Pillow==10.0.1"]
 }

--- a/homeassistant/components/matrix/manifest.json
+++ b/homeassistant/components/matrix/manifest.json
@@ -5,5 +5,5 @@
   "documentation": "https://www.home-assistant.io/integrations/matrix",
   "iot_class": "cloud_push",
   "loggers": ["matrix_client"],
-  "requirements": ["matrix-nio==0.21.2", "Pillow==10.0.1"]
+  "requirements": ["matrix-nio==0.22.0", "Pillow==10.0.1"]
 }

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -469,10 +469,6 @@ filterwarnings = [
     "ignore:'cgi' is deprecated and slated for removal in Python 3.13:DeprecationWarning:feedparser.encodings",
     # https://github.com/jaraco/jaraco.abode/commit/9e3e789efc96cddcaa15f920686bbeb79a7469e0 - update jaraco.abode to >=5.1.0
     "ignore:`jaraco.functools.call_aside` is deprecated, use `jaraco.functools.invoke` instead:DeprecationWarning:jaraco.abode.helpers.timeline",
-    # https://github.com/poljar/matrix-nio/pull/438 - >=0.22.0
-    "ignore:FormatChecker.cls_checks is deprecated:DeprecationWarning:nio.schemas",
-    # https://github.com/poljar/matrix-nio/pull/439 - >=0.22.0
-    "ignore:'cgi' is deprecated and slated for removal in Python 3.13:DeprecationWarning:nio.client.http_client",
     # https://github.com/nextcord/nextcord/pull/1095 - >2.6.1
     "ignore:pkg_resources is deprecated as an API:DeprecationWarning:nextcord.health_check",
     # https://github.com/grahamwetzler/smart-meter-texas/pull/143 - >0.5.3

--- a/requirements_all.txt
+++ b/requirements_all.txt
@@ -1183,7 +1183,7 @@ lxml==4.9.3
 mac-vendor-lookup==0.1.12
 
 # homeassistant.components.matrix
-matrix-nio==0.21.2
+matrix-nio==0.22.0
 
 # homeassistant.components.maxcube
 maxcube-api==0.4.3

--- a/requirements_all.txt
+++ b/requirements_all.txt
@@ -1183,7 +1183,7 @@ lxml==4.9.3
 mac-vendor-lookup==0.1.12
 
 # homeassistant.components.matrix
-matrix-nio==0.22.0
+matrix-nio==0.22.1
 
 # homeassistant.components.maxcube
 maxcube-api==0.4.3

--- a/requirements_test_all.txt
+++ b/requirements_test_all.txt
@@ -915,7 +915,7 @@ lxml==4.9.3
 mac-vendor-lookup==0.1.12
 
 # homeassistant.components.matrix
-matrix-nio==0.21.2
+matrix-nio==0.22.0
 
 # homeassistant.components.maxcube
 maxcube-api==0.4.3

--- a/requirements_test_all.txt
+++ b/requirements_test_all.txt
@@ -915,7 +915,7 @@ lxml==4.9.3
 mac-vendor-lookup==0.1.12
 
 # homeassistant.components.matrix
-matrix-nio==0.22.0
+matrix-nio==0.22.1
 
 # homeassistant.components.maxcube
 maxcube-api==0.4.3

